src/Components/Blazor/WebAssembly.Authentication/test/RemoteAuthenticationServiceTests.cs

```diff
@@ -0,0 +1,492 @@
+
+            // Assert
+            Assert.Equal(
+                new[] { "AuthenticationService.init", "AuthenticationService.getAccessToken" },
+                testJsRuntime.PastInvocations.Select(i => i.identifier).ToArray());
+
+            Assert.Equal(result, testJsRuntime.GetAccessTokenResult);
+            Assert.Equal(tokenOptions, (AccessTokenRequestOptions)testJsRuntime.PastInvocations[^1].args[0]);
+        }
+
+        [Fact]
+        public async Task RemoteAuthenticationService_GetUser_ReturnsAnonymousClaimsPrincipal_ForUnauthenticatedUsers()
+        {
+            // Arrange
+            var testJsRuntime = new TestJsRuntime();
+            var options = CreateOptions();
+            var runtime = new RemoteAuthenticationService<RemoteAuthenticationState, OidcProviderOptions>(
+                testJsRuntime,
+                options);
+
+            testJsRuntime.GetUserResult = null;
+
+            // Act
+            var result = await runtime.GetCurrentUser();
+
+            // Assert
+            Assert.Empty(result.Claims);
+            Assert.Single(result.Identities);
+            Assert.False(result.Identity.IsAuthenticated);
+
+            Assert.Equal(
+                new[] { "AuthenticationService.init", "AuthenticationService.getUser" },
+                testJsRuntime.PastInvocations.Select(i => i.identifier).ToArray());
+        }
+
+        [Fact]
+        public async Task RemoteAuthenticationService_GetUser_ReturnsUser_ForAuthenticatedUsers()
+        {
+            // Arrange
+            var testJsRuntime = new TestJsRuntime();
+            var options = CreateOptions();
+            var runtime = new RemoteAuthenticationService<RemoteAuthenticationState, OidcProviderOptions>(
+                testJsRuntime,
+                options);
+
+            var serializationOptions = new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase, PropertyNameCaseInsensitive = true };
+            var serializedUser = JsonSerializer.Serialize(new
+            {
+                CoolName = "Alfred",
+                CoolRole = new[] { "admin", "cool", "fantastic" }
+            }, serializationOptions);
+
+            testJsRuntime.GetUserResult = JsonSerializer.Deserialize<IDictionary<string, object>>(serializedUser);
+
+            // Act
+            var result = await runtime.GetCurrentUser();
+
+            // Assert
+            Assert.Single(result.Identities);
+            Assert.True(result.Identity.IsAuthenticated);
+            Assert.Equal("Alfred", result.Identity.Name);
+            Assert.Equal("a", result.Identity.AuthenticationType);
+            Assert.True(result.IsInRole("admin"));
+            Assert.True(result.IsInRole("cool"));
+            Assert.True(result.IsInRole("fantastic"));
+        }
+
+        [Fact]
+        public async Task RemoteAuthenticationService_GetUser_MapsScopesToRoles_IfScopeClaimIsProvided()
```

Comment:
Did Barry/Levi agree with the idea of mapping scopes to roles? I remember this being a question but can't remember if it was resolved.

Reply:
I took this out.